modules/videoio/src/cap_libv4l.cpp

```diff
@@ -1319,6 +1320,7 @@ static IplImage* icvRetrieveFrameCAM_V4L( CvCaptureCAM_V4L* capture, int) {
     switch(capture->imageProperties.palette) {
       case VIDEO_PALETTE_RGB24:
       case VIDEO_PALETTE_YUV420:
+      case VIDEO_PALETTE_YUYV:
```

Comment:
> it is not compilable without v4l2

Looks like truth.
For example, in Ubuntu file `linux/videodev2.h` is part of `linux-libc-dev` package, which is always installed on dev machine (`build-essential` depends on it). cmake sets `HAVE_CAMV4L2` flag when this file is found.